File: android-samples/poslink/EposPosLinkExample/app/src/main/java/com/example/eposappexample/poslink/ui/tables/StatusTag.kt
package com.example.eposappexample.poslink.ui.tables

import androidx.compose.runtime.Composable
import com.teya.lemonade.LemonadeUi
import com.teya.lemonade.Tag
import com.teya.lemonade.core.TagVoice
import com.teya.unifiedepossdk.poslink.models.tabs.TabStatus

@Composable
fun StatusTag(status: TabStatus) {
    val (voice, label) = when (status) {
        TabStatus.OPEN -> TagVoice.Positive to "Open"
        TabStatus.PAYING -> TagVoice.Info to "Paying"
        TabStatus.PAUSED -> TagVoice.Warning to "Paused"
        TabStatus.COMPLETED -> TagVoice.Positive to "Completed"
        TabStatus.CLOSED -> TagVoice.Neutral to "Closed"
        TabStatus.UNKNOWN -> TagVoice.Neutral to "Unknown"
    }
    LemonadeUi.Tag(label = label, voice = voice)
}

File: ios-samples/poslink/EposPosLinkExample/EposPosLinkExample/Views/Tables/StatusTag.swift
import SwiftUI
import Lemonade
import TeyaUnifiedEposSDK

struct StatusTag: View {
    let status: TeyaTabStatus

    private var label: String {
        if status == .open { return "Open" }
        if status == .paying { return "Paying" }
        if status == .paused { return "Paused" }
        if status == .completed { return "Completed" }
        if status == .closed { return "Closed" }
        return "Unknown"
    }

    private var voice: TagVoice {
        if status == .open || status == .completed { return .positive }
        if status == .paying { return .info }
        if status == .paused { return .warning }
        return .neutral
    }

    var body: some View {
        LemonadeUi.Tag(label: label, voice: voice)
    }
}

struct PaymentStatusTag: View {
    let state: TeyaPaymentState

    private var label: String {
        if state == .successful { return "Paid" }
        if state == .canceled { return "Canceled" }
        if state == .processingfailed || state == .communicationfailed { return "Failed" }
        return state.name
    }

    private var voice: TagVoice {
        if state == .successful { return .positive }
        if state == .canceled { return .neutral }
        if state == .processingfailed || state == .communicationfailed { return .critical }
        return .info
    }

    var body: some View {
        LemonadeUi.Tag(label: label, voice: voice)
    }
}
